pkg/asset/installconfig/vsphere/vsphere.go

```diff
@@ -72,17 +72,34 @@ func Platform() (*vsphere.Platform, error) {
 		return nil, errors.Wrap(err, "failed to get VIPs")
 	}
 
+	failureDomain := vsphere.FailureDomain{
+		Name:   "generated-failure-domain",
+		Zone:   "generated-zone",
+		Region: "generated-region",
+		Server: vCenter.VCenter,
+		Topology: vsphere.Topology{
+			Datacenter:     dc,
+			ComputeCluster: cluster,
+			Datastore:      datastore,
+			Networks:       []string{network},
+		},
+	}
```

Comment:
I think this belongs in https://github.com/openshift/installer/blob/master/pkg/types/vsphere/defaults/platform.go

This code would only apply to the survey right? We should be able to supply defaults in one place, so that the same code is used to supply defaults whether the survey or install config is used.

Reply:
We could generate a default failuredomain and vcenter but the only fields we could provide would be failureDomain: Name,Zone,Region and vcenters: port

Reply:
My original comment misunderstood this code. It makes more sense upon review. Thanks!